src/snowflake/snowpark/_internal/analyzer/snowflake_plan.py

```diff
@@ -976,9 +988,26 @@ def get_create_and_insert_plan(child: SnowflakePlan, replace=False, error=True):
                 propagate_referenced_ctes=False,
             )
         elif mode == SaveMode.ERROR_IF_EXISTS:
-            if is_generated:
+            if creation_source == TableCreationSource.CACHE_RESULT:
                 return get_create_and_insert_plan(child, replace=False, error=True)
 
+            if creation_source == TableCreationSource.LARGE_QUERY_BREAKDOWN:
+                return self.build(
+                    lambda x: create_table_as_select_statement(
```

Comment:
Could this be done with partial instead?

Reply:
yeah, we can. It will de-duplicate some lines in this function. Let me update that.